std/file.d

```diff
@@ -4863,20 +4864,27 @@ private struct DirIteratorImpl
     }
 }
 
-struct DirIterator
+// Must be a template, because the destructor is unsafe or safe depending on
+// whether `-preview=dip1000` is in use. Otherwise, linking errors would
+// result.
+struct DirIterator(bool useDIP1000)
```

Comment:
Unfortunately, this is a breaking change since `DirIterator` is part of the public API.

Reply:
It's not a documented symbol though, and BuildKite passing proves it doesn't break much. But yes it's possible someone still uses it.

Hmm, I wonder if this would avoid the breakage?

```D
struct _DirIterator(bool useDIP1000) {/*...*/}
alias DirIterator = _DirIterator!dip1000Enabled;
```

Previously this led to a linking error. Alias to a template instance, the compiler thinks the linker can find the function precompiled in the Phobos binary, which it can't because it was compiled with DIP1000 resulting in different ABI. But on the other hand I tested without any template parameters - will using a bool parameter for DIP1000 change the fact? I quess not, but I have to test.

Reply:
Just tested this. For some reason, doing the alias trick works fine for `DirIterator` without causing linking errors for non-dip1000 client code, *but* not for `dirEntries`. With `dirEntries` I have to keep the `useDIP1000` template parameter there. Luckily it's not so much breaking change there because functions do not require using `!()` for instantiation, except maybe if the user wants to take address of the function.

Reply:
Pushed the change for DirIterator.